packages/blocks/src/api/registration.js

```diff
@@ -24,6 +24,101 @@ import { select, dispatch } from '@wordpress/data';
+ *                                                your block, which can be
+ *                                                translated with our
+ *                                                translation functions. The
+ *                                                block inserter will show this
+ *                                                name.
+ * @property {WPBlockCategories}     [category]   Blocks are grouped into
+ *                                                categories to help users
+ *                                                browse and discover them.
+ * @property {string[]}              [keywords]   Sometimes a block could have
+ *                                                aliases that help users
+ *                                                discover it while searching.
+ *                                                For example, an `image` block
+ *                                                could also want to be
+ *                                                discovered by `photo`. You can
+ *                                                do so by providing an array of
+ *                                                terms (which can be
+ *                                                translated).
+ * @property {WPBlockTypeIconRender} [icon]       An icon property should be
+ *                                                specified to make it easier to
+ *                                                identify a block. These can be
+ *                                                any of the WordPress
+ *                                                Dashicons, or a custom `svg`
+ *                                                element.
+ * @property {WPBlockAttributes}     [attributes] Attributes provide the
+ *                                                structured data needs of a
+ *                                                block. They can exist in
+ *                                                different forms when they are
+ *                                                serialized, but they are
+ *                                                declared together under a
+ *                                                common interface.
+ * @property {WPBlockStyle[]}        [style]      Block styles can be used to
+ *                                                provide alternative styles to
+ *                                                block. It works by adding a
+ *                                                class name to the block’s
+ *                                                wrapper. Using CSS, a theme
+ *                                                developer can target the class
+ *                                                name for the style variation
+ *                                                if it is selected.
+ * @property {Function}              edit         TODO: ...
```

Comment:
I see that this is marked todo, but [see here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__blocks/index.d.ts#L69) for what I'd recommend typing this as....

(same for `save` below)

Reply:
I guess I should say that I do understand that generics aren't as well supported in JSDoc, but `ComponentType<any>` should work at the very least!